### PR TITLE
(CDAP-15715) Added support for PEM cert file for router service

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Review by anwar6953 — APPROVED** [review]

File: cdap-common/src/main/java/io/cdap/cdap/common/security/HttpsEnabler.java [hunk 1]
```diff
@@ -173,25 +174,31 @@ public HttpsURLConnection enable(HttpsURLConnection urlConn) {
...
-      if (hasTrustManager) {
-        contextBuilder = contextBuilder.trustManager(tmf);
-      }
+  /**
+   * Returns a {@link SSLHandlerFactory} based on the configuration in this class.
+   */
+  public SSLHandlerFactory createSSLHandlerFactory() {
+    KeyManagerFactory kmf = keyManagerFactory;
+    if (kmf == null) {
+      throw new IllegalArgumentException("Missing keystore to create SslContext for netty http server.");
```
Comment (CuriousVini) [inline cdap-common/src/main/java/io/cdap/cdap/common/security/HttpsEnabler.java:190]: nit: should add `@throws` in the javadoc for `IllegalArgumentException`

Comment (chtyim) [reply cdap-common/src/main/java/io/cdap/cdap/common/security/HttpsEnabler.java:190]: Added

**Review by CuriousVini — APPROVED** [review]
lgtm 👍

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-DUT7057

**Pull request merged**